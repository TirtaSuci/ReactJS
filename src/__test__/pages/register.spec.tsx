// âœ… Must be at the very top of the file, BEFORE importing RegisterPage
jest.mock("next/router", () => ({
    useRouter: () => ({
        route: "/auth/register",
        pathname: "/",
        query: {},
        asPath: "/auth/register",
        push: jest.fn(),
        replace: jest.fn(),
        reload: jest.fn(),
        back: jest.fn(),
        prefetch: jest.fn().mockResolvedValue(undefined),
        isFallback: false,
        events: {
            on: jest.fn(),
            off: jest.fn(),
            emit: jest.fn(),
        },
    }),
}));

// Only now do you import the component that uses useRouter
import RegisterPage from "@/pages/auth/register";
import { render, screen } from "@testing-library/react";

describe("Register Page", () => {
    it("should render correctly", () => {
        const { asFragment } = render(<RegisterPage />);
        expect(asFragment()).toMatchSnapshot();
    });
});
