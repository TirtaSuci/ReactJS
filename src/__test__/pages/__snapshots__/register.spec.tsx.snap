// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="register"
    >
      <p
        class="register__title"
      >
        Register Page
      </p>
      <div
        class="register__form"
      >
        <form
          class="register__form__content"
        >
          <label
            for="username"
          >
            Username
          </label>
          <input
            class="register__form__content__input"
            id="username"
            name="username"
            required=""
            type="text"
          />
          <label
            for="email"
          >
            Email
          </label>
          <input
            class="register__form__content__input"
            id="email"
            name="email"
            required=""
            type="email"
          />
          <label
            for="password"
          >
            Password
          </label>
          <div
            class="passwordWrapper"
          >
            <input
              class="inputWithToggle"
              id="password"
              name="password"
              required=""
              type="password"
            />
            <button
              class="togglePassword"
              type="button"
            >
              👁️
            </button>
          </div>
          <label
            for="confirmPassword"
          >
            Confirm Password
          </label>
          <div
            class="passwordWrapper"
          >
            <input
              class="inputWithToggle"
              id="confirmPassword"
              name="confirmPassword"
              required=""
              type="password"
            />
            <button
              class="togglePassword"
              type="button"
            >
              👁️
            </button>
          </div>
          <button
            class="register__form__content__button"
            type="submit"
          >
            Register
          </button>
        </form>
        <p
          class="register__form__login"
        >
          Sudah punya akun? 
          <a
            class="register__form__login__link"
            href="/auth/login"
          >
            Login disini
          </a>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
